modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :draft,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged_at,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :review_comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :additions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :deletions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :changed_files_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :labels,
+                 setter: ->(*) { nil }
+
+        property :github_user,
+                 getter: ->(*) {
+                   {
+                     login: github_user.github_login,
+                     htmlUrl: github_user.github_html_url,
+                     avatarUrl: github_user.github_avatar_url
+                   }
+                 },
+                 setter: ->(*) { nil }
+
+        property :merged_by,
+                 getter: ->(*) {
+                   next unless merged_by
+
+                   {
+                     login: merged_by.github_login,
+                     htmlUrl: merged_by.github_html_url,
+                     avatarUrl: merged_by.github_avatar_url
+                   }
+                 },
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :github_check_runs,
+                 getter: ->(*) {
+                   latest_check_runs.map do |check_run|
+                     {
+                       htmlUrl: check_run.github_html_url,
+                       appOwnerAvatarUrl: check_run.github_app_owner_avatar_url,
+                       name: check_run.name,
+                       status: check_run.status,
+                       conclusion: check_run.conclusion,
+                       outputTitle: check_run.output_title,
+                       outputSummary: check_run.output_summary,
+                       detailsUrl: check_run.details_url,
+                       startedAt: check_run.started_at&.iso8601,
+                       completedAt: check_run.completed_at&.iso8601
+                     }
+                   end
+                 },
+                 setter: ->(*) { nil }
+
+        date_time_property :created_at
+
+        date_time_property :updated_at
+
+        def _type
+          'GithubPullRequest'
+        end
+
+        self.to_eager_load = %i[github_user
+                                merged_by
+                                github_check_runs]
```

Comment:
`github_check_runs` cannot be eager loaded. This method may be used to do something like this in case a collection is rendered:
```
GithubPullRequest.include(GithubPullRequestRepresenter.to_eager_load)
```
which will fail.

Reply:
yes, does not make much sense when we need the special query to fetch them anyways 👍  thanks